subprojects/logging/src/main/java/org/gradle/internal/logging/console/ProgressOperation.java

```diff
@@ -65,7 +65,21 @@ public ProgressOperation getParent() {
         return parent;
     }
 
-    public Set<ProgressOperation> getChildren() {
-        return children;
+    public boolean addChild(ProgressOperation operation) {
+        if (children == null) {
```

Comment:
We rely on upstream synchronization from ThrottlingOutputEventListener calling onOutput() 

Since these progress operations are only managed in WorkInProgressRenderer, I don't think there's any thread safety concerns.